### ISSUE TITLE
Store warnings in externs?

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#3198]
Maybe this was discussed before, I can't remember. Just thinking it'd be nice to have, since then warnings could always be shown regardless of whether the build is incremental or not.

I know `-psa` has `--stash` to implement this on top of the compiler interface, but it's not completely reliable, so by storing in the externs we'd only have one set of cache invalidation related bugs to deal with. :wink:

**Comment by paf31** [comment]
Why is it desirable to display warnings generated from modules you didn't compile though? For a batch mode compiler, that seems kind of odd to me.

**Comment by garyb** [comment]
Yeah, maybe, I dunno. It’s just tedious having to wait a couple of minutes to get the warnings back every time.

The warnings are still relevant whether the files were changed or not too - the state that the project is in is still questionable.

**Comment by kRITZCREEK** [comment]
Also see #2308
